Symfony 3 changes

timwhite opened the pull request:
Not sure exactly which version it became required to do it this way, but certainly Symfony 3 forces you to use this new method of doing things.
These are my changes to get it working (Although I haven't fixed DocumentHiddenType)

timwhite committed:
Changes required for Symfony 3

timwhite committed:
Update README to reflect the Symfony 3 changes

Glifery commented:
Thanks @timwhite. Great work!
I've reviewed your changes, all looks good. But I think there must be a couple of fixes before we could merge it.

At first, we should change required Symfony version in `composer.json` file. It couldn't be `>=2.3` anymore because we need a `3.*` Symfony version support. But there is a problem with `configureOptions` and `setDefaultOptions` methods compatibility.
1. The `configureOptions` method was introduced only since `2.7` version (as mentioned [here](http://symfony.com/doc/2.7/cookbook/form/create_custom_field_type.html))
2. The `setDefaultOptions` method was removed in `3.0` version.

So we have two different methods for `<2.7` and `>3.0` Symfony versions. It means that your changes is not compatible with `<2.7`. Maybe we should separate your pull request to another branch and tag it as `for 3.* only`? Any ideas?

And secondary, It would be cool to add the same changes into `DocumentHiddenType` for users who expect to get a `3.*` version compatibility for whole functionality. I think it wouldn't take much time.

pivolan commented:
I think,  it is a good idea to separate pull request.

pivolan committed:
Symfony3 support.
Removed deprecated getName methods.
removed alias from services.
Set requirement symfony3.0

apourzand commented:
Hi, I forked the latest version and made changes to make it compatible with symfony 3. I am using it in my sf3 project and it works well. However, being new to Github I don't know What is the correct way to contribute. So please feel free to visit the forked version for review. Great job by the way.

mariusz-kraj-c commented:
+1

speto commented:
Is this symfony 3 support stucked? I would like to move on from @pivolan fork and I think the next possible steps to solve this could be:
1. @Glifery should create a new upstream branch for symfony 3 support
2. @timwhite should merge changes from @pivolan https://github.com/timwhite/EntityHiddenTypeBundle/pull/1 and change this pull request target to the new upstream branch for symfony 3 support
3. @Glifery should then merge @timwhite and @pivolan work to symfony 3 support branch

If there will be issue with symfony 3 we can fill it out and continue work on it in separate branch.

Finally there could be a way to merge both versions and provide support for both `~2.6|~3.0` like I saw e.g. in https://github.com/braincrafted/bootstrap-bundle

Glifery commented:
Okay guys, lets do it! :)
I've created a branch [symfony-3-support](https://github.com/Glifery/EntityHiddenTypeBundle/tree/symfony-3-support) so @timwhite welcome to accept @pivolan 's changes

timwhite committed:
Merge pull request #1 from pivolan/master

Symfony3 support for Document.

timwhite commented:
@Glifery @speto 
Sorry guys. Been busy with other work related stuff. Not sure if merging @pivolan's request then changing the source branch was all that was needed. Let me know if you wanted more and I'll try to get to it this weekend.

mariusz-kraj-c commented:
@timwhite  Hi, when you planning to merge this changes?

timwhite commented:
@mariusz-kraj-c I think I've done what's needed. Waiting on @Glifery to merge it into upstream I think? Someone correct me if I'm wrong and more from me was needed.
